protected/humhub/modules/user/behaviors/Followable.php

```diff
@@ -143,37 +143,59 @@ public function isFollowedByUser($userId = null, $withNotifications = false)
         return false;
     }
 
+    /**
+     * Get a query of users which are followers of this object.
+     *
+     * @param ActiveQueryUser $query e.g. for limit the result
+     * @return ActiveQueryUser
+     */
+    public function getFollowersQuery($query = null)
+    {
+        if ($query === null) {
+            $query = User::find();
+        }
+
+        return $query->visible()
+            ->leftJoin('user_follow', 'user.id = user_follow.user_id AND user_follow.object_id=:object_id AND user_follow.object_model = :object_model', [
+                ':object_model' => get_class($this->owner),
+                ':object_id' => $this->owner->getPrimaryKey(),
+            ])
+            ->andWhere('user_follow.user_id IS NOT null');
+    }
+
+    /**
+     * Get a query of users which are followers with enabled notifications of this object.
+     *
+     * @return ActiveQueryUser
+     */
+    public function getFollowersWithNotificationQuery()
+    {
+        return $this->getFollowersQuery()
+            ->andWhere('user_follow.send_notifications=1');
+    }
+
     /**
      * Returns the number of users which are followers of this object.
      *
      * @return int
      */
     public function getFollowerCount()
     {
-        return Follow::find()->where(['object_id' => $this->owner->getPrimaryKey(), 'object_model' => $this->owner->className()])->count();
+        return $this->getFollowersQuery()->count();
     }
 
     /**
      * Returns an array of users which are followers of this object.
      *
-     * @param CDbCriteria $eCriteria e.g. for limit the result
+     * @deprecated Use $this->getFollowersQuery()->all() instead
+     * @param ActiveQueryUser $query e.g. for limit the result
      * @param boolean $withNotifications only return followers with enabled notifications
      * @param boolean $returnQuery only return the query instead of User objects
-     * @return Users[]|ActiveQueryUser the user objects or the active query
+     * @return User[]|ActiveQueryUser the user objects or the active query
```

Comment:
We should deprecate this one.

Reply:
> We still should remove the $query param here.

Commit https://github.com/humhub/humhub/pull/4879/commits/5394e00c89fce92a6220d916d5a555a5f127fb8e.